jax_unirep/evotuning.py

```diff
@@ -82,39 +81,86 @@ def avg_loss(
     sum_loss = 0
     num_seqs = 0
     global evotune_loss  # this is necessary for JIT to reference evotune_loss
-    evotune_loss_jit = jit(evotune_loss, backend=backend)
-
-    def batch_iter(
-        xs: np.ndarray, ys: np.ndarray, batch_size: int = batch_size
-    ):
-        i = 0
-        for i in tqdm(
-            range(0, len(xs), batch_size),
-            desc=f"Avg loss on dataset length {len(xs)}",
-        ):
+    evotune_loss_jit = jit(
+        partial(evotune_loss, predict=model_func), backend=backend
+    )
+
+    def batch_iter(xs: np.ndarray, ys: np.ndarray, batch_size: int):
+        for i in range(0, len(xs), batch_size):
             yield xs[i : i + batch_size], ys[i : i + batch_size]
-            i += batch_size
 
     for xmat, ymat in zip(xs, ys):
-        # Send x and y in small batches of 100 to control memory usage.
+        # Send x and y in small batches to control memory usage.
         for x, y in batch_iter(xmat, ymat, batch_size=batch_size):
-            sum_loss += evotune_loss_jit(params, inputs=x, targets=y) * len(x)
+            sum_loss += evotune_loss_jit(
+                params=params, inputs=x, targets=y
+            ) * len(x)
             num_seqs += len(x)
 
     return sum_loss / num_seqs
 
 
+def generate_one_length_batch(
```

Comment:
```suggestion
def generate_single_length_batch(
```

Maybe this describes the purpose of the function a bit better?